testsuite/features/step_definitions/common_steps.rb

```diff
@@ -543,11 +543,13 @@
 end
 
 When(/^I click promote from Development to QA$/) do
-  raise 'Click on promote from Development failed' unless find_button('dev_label-promote-modal-link', disabled: false, wait: DEFAULT_TIMEOUT).click
+  promote_first = find_all(:xpath, "//button[contains(., 'Promote')]")[0]
```

Comment:
That could fail. And it will not be properly handled.
If the find_all have no matches, it will throw an error, it will not caught, and the next line bellow will not run.

Reply:
Right, there should be a check on the existence of the button first, I'll add it